### PR TITLE
Make identity naming more consistent + fix bugs

### PR DESCRIPTION
**Pull request by jdetter**
# Description of Changes

Please describe your change, mention any related tickets, and so on here.

- There are several places in the CLI where you can set the name of an identity:
  - `spacetime identity new --name <name> --no-email`
  - `spacetime identity set-name <identity> <name>`
  - `spacetime identity import <identity> <token> --name <name>`
 
In these places we had slightly different logic for validating that an identity name looks okay. I've made the logic in these places more consistent. This will prevent people from accidentally adding identities with the same name or with an invalid name (identity names cannot look like identities).

# API and ABI breaking changes

If this is an API or ABI breaking change, please apply the
corresponding GitHub label.

Not breaking

# Expected complexity level and risk

*How complicated do you think these changes are? Grade on a scale from 1 to 5,
where 1 is a trivial change, and 5 is a deep-reaching and complex change.*

*This complexity rating applies not only to the complexity apparent in the diff,
but also to its interactions with existing and future code.*

*If you answered more than a 2, explain what is complex about the PR,
and what other components it interacts with in potentially concerning ways.*

0

# Testing

*Describe any testing you've done, and any testing you'd like your reviewers to do,
so that you're confident that all the changes work as expected!*

- [x] *Write a test you've completed here.*
- [x] Create a new identity with the name admin. Then create another identity with no name. Export this new identity. Delete the identity. Now import this identity with the name admin, you will get an error.

### CODE REVIEW

File: crates/cli/src/subcommands/identity.rs
Comment (bfops) [inline crates/cli/src/subcommands/identity.rs:580]: This is a regression. It essentially renders unnamed identities unusable.
```
   $ spacetime identity new -s local --no-email | tee output
    IDENTITY     631d3896d10a5fa9db73a95a5a2e20981ab71bc32cb908859e449b5eeb886e9a
    NAME         
    EMAIL        
 
   $ cat output | grep 'IDENTITY' | awk '{print $2;}' | tee identity
   631d3896d10a5fa9db73a95a5a2e20981ab71bc32cb908859e449b5eeb886e9a
 
   $ spacetime identity set-name $(cat identity) new-identity-name
-  Updated identity:
-   IDENTITY  631d3896d10a5fa9db73a95a5a2e20981ab71bc32cb908859e449b5eeb886e9a
-   NAME      new-identity-name
+  Error: An identity name cannot be an identity.
+  [1]
```

**Review by bfops — CHANGES_REQUESTED** [review]

File: crates/cli/src/subcommands/identity.rs
Comment (jdetter) [reply crates/cli/src/subcommands/identity.rs]: Thank you for catching this Zeke that would have been a disaster 😦

File: crates/cli/src/subcommands/identity.rs
Comment (jdetter) [inline crates/cli/src/subcommands/identity.rs:452]: I've also added a check here to prevent adding the same identity multiple times

File: crates/cli/src/util.rs
Comment (bfops) [inline crates/cli/src/util.rs:154]: why did we need this check? shouldn't `get_identity_config("")` fail anyway?

File: crates/cli/src/config.rs
Comment (bfops) [inline crates/cli/src/config.rs:904]: how would you feel about making this like:
```rs
pub fn can_set_name(&self, new_nickname: &str) -> Result<(), anyhow::Error> {
```
(it seems a bit weird to call `can_set_name(None)`?)

File: crates/cli/src/subcommands/publish.rs
Comment (bfops) [inline crates/cli/src/subcommands/publish.rs:116]: could we just move the later code to here, rather than effectively duplicating this check? like could we put this up here:
```
    let (auth_header, identity) = get_auth_header(&mut config, anon_identity, identity, server)
        .await?
        .unzip();
```

File: crates/cli/src/util.rs
Comment (bfops) [inline crates/cli/src/util.rs:155]: (the extra `""` make it clearer when it's just the empty string)

File: crates/cli/src/subcommands/publish.rs
Comment (bfops) [inline crates/cli/src/subcommands/publish.rs:121]: moving this earlier means that we check `identity` validity before we go through a (potentially very long) build.

File: crates/cli/src/subcommands/identity.rs
Comment (bfops) [reply crates/cli/src/subcommands/identity.rs:456]: (factored out `config.identity_exists` so it's somewhat self-describing)

**Review by bfops — APPROVED** [review]
LGTM once the base PR merges

**Pull request merged**
